sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessCreate.java

```diff
@@ -204,6 +209,32 @@ public T getCurrent() throws NoSuchElementException {
 
       @Override
       public void close() throws IOException {}
+
+      @Override
+      public Double getFractionConsumed() {
+        return rangeTracker.getFractionConsumed();
+      }
+
+      @Override
+      public InMemorySource<T> splitAtFraction(double fraction) {
+        long splitPosition = rangeTracker.getPositionForFractionConsumed(fraction);
+        if (rangeTracker.trySplitAtPosition(splitPosition)) {
+          List<byte[]> oldSourceElems =
+              ImmutableList.copyOf(source.allElementsBytes.subList(0, (int) splitPosition));
+          List<byte[]> newSourceElems =
+              ImmutableList.copyOf(
+                  source.allElementsBytes.subList(
```

Comment:
I'd consider seeing if you can simplify this logic. BigtableIO is doing a lot more complex operation here but it manages to have simpler, more readable code :)

https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/bigtable/BigtableIO.java#L776